capstone/src/test/java/com/google/sps/servlets/UpdateLikesServletTest.java

```diff
@@ -0,0 +1,166 @@
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import com.google.sps.Objects.Post;
+import com.google.sps.Objects.Comment;
+
+public class UpdateLikesServletTest {
+
+  private static final String USER_EMAIL = "test@test.com";
+  private static final String USER_ID = "test";
+  private static final String AUTHOR_ID = "123123123";
+  private static final String POST_TEXT = "a great post";
+  private static final String CHALLENGE_NAME = "run 4 miles";
+  private static final String IMG = "";
+  private static final long TIMESTAMP = 123123123;
+  private static final long POST_ID = 1;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig()
+                  .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+              new LocalUserServiceTestConfig())
+          .setEnvEmail(USER_EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail.com")
+          .setEnvAttributes(
+              new HashMap(
+                  ImmutableMap.of(
+                      "com.google.appengine.api.users.UserService.user_id_key", USER_ID)));
+
+  private final Post POST_1 = new Post(
+    POST_ID, /* postId */ 
+    AUTHOR_ID, /* authorId */ 
+    POST_TEXT, /* postText */
+    new ArrayList<Comment>(), /* comments */
+    CHALLENGE_NAME, /* challengeName */
+    TIMESTAMP, /* timestamp */
+    IMG, /* img */
+    new HashSet<String>() /* likes */);
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private DatastoreService datastore;
+  private UpdateLikesServlet updateLikesServlet;
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Set up a fake HTTP response.
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+    updateLikesServlet = new UpdateLikesServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+    responseWriter = null;
+    datastore = null;
+    updateLikesServlet = null;
+  }
+
+  @Test
+  public void doPost_userNotLoggedIn() throws IOException, EntityNotFoundException {
+    helper.setEnvIsLoggedIn(false);
+    when(mockRequest.getParameter("id")).thenReturn(Long.toString(POST_ID));
+    when(mockRequest.getParameter("liked")).thenReturn("true");
+
+    updateLikesServlet.doPost(mockRequest, mockResponse);
+    String response = responseWriter.toString();
+    assertTrue(response.contains("Oops an error happened!"));
+  }
+
+  @Test
+  public void doPost_invalidPost() throws IOException, EntityNotFoundException {
+    populateDatabase(datastore);
```

Comment:
You should call this once in the setUp instead of at the beginning of every test